### PR TITLE
[BEAM-6300] Put generated getter/setter/creator classes in the same package as the class they access

### PR DESCRIPTION
**Pull request by reuvenlax**
This allows the user to use package-private classes and fields.

R: @kanterov 

This has already been reviewed. Splitting into a separate PR to ease bisection/rollback.

### CODE REVIEW

**Comment by reuvenlax** [comment]
Run Java PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/ByteBuddyUtils.java [hunk 1]
```diff
@@ -59,6 +62,40 @@
   private static final ForLoadedType READABLE_INSTANT_TYPE =
       new ForLoadedType(ReadableInstant.class);
 
+  /**
+   * A naming strategy for ByteBuddy classes.
+   *
+   * <p>We always inject the generatter classes in the same same package as the user's target class.
+   *
+   * @kanterov kanterov 20 hours ago Contributor nit: s/generatter/generated/
```
Comment (kanterov) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/ByteBuddyUtils.java]: not a big deal, let's fix it later

Comment (kanterov) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/ByteBuddyUtils.java]: probably in https://github.com/apache/beam/pull/7334

**Comment by reuvenlax** [comment]
Yeah, saw that too. already one it.

**Pull request merged**